plugins/processors/ec2tagger/constants.go

```diff
@@ -0,0 +1,93 @@
+  ## If this configuration contains one entry and its value is "*", then ALL EC2 Instance Tags for the instance are applied.
+  ## Note: This plugin renames the "aws:autoscaling:groupName" EC2 Instance Tag key to be spelled "AutoScalingGroupName".
+  ## This aligns it with the AutoScaling dimension-name seen in AWS CloudWatch.
+  # ec2_instance_tag_keys = ["aws:autoscaling:groupName", "Name"]
+  ##
+  ## Retrieve ebs_volume_id for the specified devices, add ebs_volume_id as tag. The specified devices are
+  ## the values corresponding to the tag key "disk_device_tag_key" in the input metric.
+  ## If this configuration is not provided, or has an empty list, no ebs volume is applied.
+  ## If this configuration contains one entry and its value is "*", then all ebs volume for the instance are applied.
+  # ebs_device_keys = ["/dev/xvda", "/dev/nvme0n1"]
+  ##
+  ## Specify which tag to use to get the specified disk device name from input Metric
+  # disk_device_tag_key = "device"
+  ##
+  ## Amazon Credentials
+  ## Credentials are loaded in the following order
+  ## 1) Assumed credentials via STS if role_arn is specified
+  ## 2) explicit credentials from 'access_key' and 'secret_key'
+  ## 3) shared profile from 'profile'
+  ## 4) environment variables
+  ## 5) shared credentials file
+  ## 6) EC2 Instance Profile
+  # access_key = ""
+  # secret_key = ""
+  # token = ""
+  # role_arn = ""
+  # profile = ""
+  # shared_credential_file = ""
+`
+
+const (
+	ec2InstanceTagKeyASG = "aws:autoscaling:groupName"
+	cwDimensionASG       = "AutoScalingGroupName"
+	mdKeyInstanceId      = "InstanceId"
+	mdKeyImageId         = "ImageId"
+	mdKeyInstaneType     = "InstanceType"
+	ebsVolumeId          = "EBSVolumeId"
+)
+
+const (
+	metadataCheckStrStartInitialization         = "ec2tagger: EC2 IMDS has started initialization."
+	metadataCheckStrTagNotSupported             = "ec2tagger: Unsupported EC2 Metadata key: %s."
+	metadataCheckStrInstanceDocumentFailure     = "ec2tagger: Unable to retrieve Instance Metadata Tags."
+	metadataCheckStrEC2InstanceTagger           = "This plugin must only be used on an EC2 instance."
+	metadataCheckStrIncreaseHopLimit            = "ec2tagger: Please increase hop limit to 3 by following this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-options.html#configuring-IMDS-existing-instances."
+	ec2TagAndVolumeCheckStrInitRetrievalSuccess = "ec2tagger: Initial retrieval of tags succeeded"
+	ec2TagAndVolumeCheckStrStartInitialization  = "ec2tagger: EC2 tagger has started initialization."
+	ec2VolumeCheckStrInitRetrievalFailure       = "ec2tagger: Unable to describe ec2 volume for initial retrieval: %v"
+	ec2TagCheckStrInitRetrievalFailure          = "ec2tagger: Unable to describe ec2 tags for initial retrieval: %v"
+	ec2TagAndVolumeCheckStrStartRefresh         = "ec2tagger refreshing: EC2InstanceTags needed %v, retrieved: %v, ebs device needed %v, retrieved: %v"
```

Comment:
I still disagree with these constant strings for logs, but will defer to others that review this PR.

Reply:
Agree 👍 .

Reply:
I just looked, and they are **NOT** constant strings that are used for log messages. Can you please explain your rationale for correlating their usage of constant strings with our log messages?

Reply:
Wdym @SaxyPandaBear?  The SSM Agent creates these constant strings that are used for log messages as shown below and there are [another boiler plate for create constant as a separate file](https://github.com/aws/amazon-ssm-agent/blob/mainline/agent/appconfig/constants.go)
```
const (
	maxTimeCallMetadataSeconds = 3
	metadataCheckStrName              = "EC2 IMDS"
	metadataCheckStrInstanceIdTimeout = "instance id request timeout"
	metadataCheckStrInstanceIdFailure = "failed to query instance id from IMDS: %v"
	metadataCheckStrRegionTimeout     = "region request timeout"
	metadataCheckStrRegionFailure     = "failed to query region from IMDS: %v"
	metadataCheckStrHasHybrid         = "Instance has hybrid registration"
	metadataCheckStrIMDSTimeout       = "IMDS request timeout"
	metadataCheckStrIMDSFailure       = "Failed to query IMDS: %v"
	metadataCheckStrSuccess           = "IMDS is accessible and has instance id %s in region %s"
)
```

Reply:
Can you please identify specific lines of code where those constants are used for log messages? They are all used either for error strings or return values.

Reply:
None of the constants in https://github.com/aws/amazon-ssm-agent/blob/mainline/agent/appconfig/constants.go look like they are log message constants either.

Reply:
I'm not sure I am understand your point, what is the difference between log messages using constant and [return note message with constant](https://github.com/aws/amazon-ssm-agent/blob/mainline/agent/cli/diagnostics/metadata_check.go#L161)?

Reply:
The second one is the boiler plate to create a separate constant string file, it does not relate to use constant log message.

Reply:
Six months from now, if someone is debugging an issue with the ec2tagger and needs to look through the logs, it will take longer to trace through the behavior of the plugin because the log messages are pulled out into a separate constants file. 

> They are using the same idea of using constant with log message

They are **not** the same. Reusing a constant string can make sense when it is a frequently used parameter. But these are diagnostic log messages. We should be able to very quickly identify where in the code that these log messages are getting emitted, and putting the log strings into a different file makes the log messages unreadable in the ec2tagger plugin source code. 

---
### Example
Pretend that we get a customer ticket and we see this when running on an EC2 instance:
```
2022-06-07T03:36:11Z E! [processors.ec2tagger] ec2tagger: Unable to retrieve Instance Metadata Tags. This plugin must only be used on an EC2 instance.
2022-06-07T03:36:11Z E! [processors.ec2tagger] ec2tagger: Please increase hop limit to 3 by following this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-options.html#configuring-IMDS-existing-instances.
```
The above log lines correspond to this in the code:
```go
	if err != nil {
		t.Log.Errorf(metadataCheckStrInstanceDocumentFailure)
		if context.CurrentContext().RunInContainer() {
			t.Log.Errorf(metadataCheckStrIncreaseHopLimit)
		}
		return err
	}
```
I only know that because I am reviewing this PR today. What are the steps that someone six months or a year from now would take in order to get to this part of the code? They would have to search the project for the contents of the log message, e.g.: `Unable to retrieve Instance Metadata Tags`, and that takes them to this constants file. 
```go
metadataCheckStrInstanceDocumentFailure = "ec2tagger: Unable to retrieve Instance Metadata Tags. This plugin must only be used on an EC2 instance."
```
This line alone does not provide **any** context at which point this log line gets emitted. Now the person looking into the issue has to search through the project **again** for where this variable `metadataCheckStrInstanceDocumentFailure` is used. 

Not every string needs to be a constant. Being a constant does not automatically make it cleaner or more readable/maintainable.

Reply:
👍  Agree. Thought about it at some point but in exchange for readable. Sorry for this change and I have reverted back with normal log string message.